Fixes 8327

iwyoo opened the pull request:
<!--
Thanks for contributing a pull request! 
-->
#### Reference Issue
Fixes 8327

#### What does this implement/fix? Explain your changes.
This applies rootedAtAtom correctly for multiple fragments

```python
from rdkit import Chem

smiles = "[C:1][C:2].[N:3]([C:4])=[O:5]"
mol = Chem.MolFromSmiles(smiles)
for i in range(5):
    print(f"Starting at atom {i + 1}: {Chem.MolToSmiles(mol, rootedAtAtom=i)}")
```
```
Starting at atom 1: [C:1][C:2].[N:3]([C:4])=[O:5]
Starting at atom 2: [C:2][C:1].[N:3]([C:4])=[O:5]
Starting at atom 3: [C:1][C:2].[N:3]([C:4])=[O:5]
Starting at atom 4: [C:1][C:2].[C:4][N:3]=[O:5]
Starting at atom 5: [C:1][C:2].[O:5]=[N:3][C:4]
```

iwyoo commented:
https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/SmilesParse/catch_tests.cpp#L2777
This part failed the test.

https://github.com/rdkit/rdkit/pull/7811
In this PR, the failed test is added. However, I think the test should target MolFragmentToSmiles (not MolToSmilse?). I'm not sure since I don't know the context exactly.

Although it wasn't intentional, should this PR also cover MolFragmentToSmiles?

greglandrum commented:
@iwyoo Thanks for the contribution!

> https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/SmilesParse/catch_tests.cpp#L2777 This part failed the test.
> 
> #7811 In this PR, the failed test is added. However, I think the test should target MolFragmentToSmiles (not MolToSmilse?). I'm not sure since I don't know the context exactly.
> 
> Although it wasn't intentional, should this PR also cover MolFragmentToSmiles?
>
> https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/SmilesParse/catch_tests.cpp#L2777 This part failed the test.
> 
> #7811 In this PR, the failed test is added. However, I think the test should target MolFragmentToSmiles (not MolToSmilse?). I'm not sure since I don't know the context exactly.
> 
> Although it wasn't intentional, should this PR also cover MolFragmentToSmiles?

Your change (which looks good, by the way) causes the test for issue #7572 in the smiles tests to fail because the test is expecting an error from `MolToSmiles()` that is no longer raised - your fix means the error does not need to be raised... this is good!
I would suggest you update the test to no longer expect the test to fail and to check the SMILES that is now returned for correctness.

We also try to always add tests for bugs we fix or features we add, so please add a test to the bottom of `Code/GraphMol/SmilesParse/catch_tests.cpp` to explicitly demonstrate that rootedAtAtom now works if there are multiple fragments. Please include a couple of examples that show that providing a `rootedAtAtom` value in each of the fragments returns correct results.

iwyoo commented:
@greglandrum I changed the old test and added other examples. Canonicalization has the property of sorting fragments separately from rootedAtAtom, which seems to be the responsibility of the user.

greglandrum reviewed (approved):
LGTM

greglandrum commented:
Thanks for the contribution @iwyoo !
